platform/net/photonos_net_manager.go

```diff
@@ -0,0 +1,230 @@
+	ipnet "net"
+	"strconv"
+	"strings"
+)
+
+const photonosNetManagerLogTag = "photonosNetManager"
+
+type photonosNetManager struct {
+	fs                            boshsys.FileSystem
+	cmdRunner                     boshsys.CmdRunner
+	routesSearcher                RoutesSearcher
+	ipResolver                    boship.Resolver
+	interfaceConfigurationCreator InterfaceConfigurationCreator
+	interfaceAddressesValidator   boship.InterfaceAddressesValidator
+	dnsValidator                  DNSValidator
+	addressBroadcaster            bosharp.AddressBroadcaster
+	logger                        boshlog.Logger
+}
+
+func NewPhotonosNetManager(
+	fs boshsys.FileSystem,
+	cmdRunner boshsys.CmdRunner,
+	ipResolver boship.Resolver,
+	interfaceConfigurationCreator InterfaceConfigurationCreator,
+	interfaceAddressesValidator boship.InterfaceAddressesValidator,
+	dnsValidator DNSValidator,
+	addressBroadcaster bosharp.AddressBroadcaster,
+	logger boshlog.Logger,
+) Manager {
+	return photonosNetManager{
+		fs:                            fs,
+		cmdRunner:                     cmdRunner,
+		ipResolver:                    ipResolver,
+		interfaceConfigurationCreator: interfaceConfigurationCreator,
+		interfaceAddressesValidator:   interfaceAddressesValidator,
+		dnsValidator:                  dnsValidator,
+		addressBroadcaster:            addressBroadcaster,
+		logger:                        logger,
+	}
+}
+
+func (net photonosNetManager) SetupIPv6(_ boshsettings.IPv6, _ <-chan struct{}) error { return nil }
+
+func (net photonosNetManager) SetupNetworking(networks boshsettings.Networks, errCh chan error) error {
+	nonVipNetworks := boshsettings.Networks{}
+	for networkName, networkSettings := range networks {
+		if networkSettings.IsVIP() {
+			continue
+		}
+		nonVipNetworks[networkName] = networkSettings
+	}
+
+	staticInterfaceConfigurations, dhcpInterfaceConfigurations, err := net.buildInterfaces(nonVipNetworks)
+	if err != nil {
+		return err
+	}
+
+	dnsNetwork, _ := nonVipNetworks.DefaultNetworkFor("dns")
+	dnsServers := dnsNetwork.DNS
+
+	err = net.writeNetworkInterfaces(dhcpInterfaceConfigurations, staticInterfaceConfigurations, dnsServers)
+	if err != nil {
+		return bosherr.WrapError(err, "Writing network configuration")
+	}
+
+	net.restartNetworkingInterfaces()
+
+	staticAddresses, dynamicAddresses := net.ifaceAddresses(staticInterfaceConfigurations, dhcpInterfaceConfigurations)
+
+	err = net.interfaceAddressesValidator.Validate(staticAddresses)
+	if err != nil {
+		return bosherr.WrapError(err, "Validating static network configuration")
+	}
+
+	err = net.dnsValidator.Validate(dnsServers)
+	if err != nil {
+		return bosherr.WrapError(err, "Validating dns configuration")
+	}
+
+	net.broadcastIps(append(staticAddresses, dynamicAddresses...), errCh)
+
+	return nil
+}
+
+func (net photonosNetManager) GetConfiguredNetworkInterfaces() ([]string, error) {
+	interfaces := []string{}
+
+	interfacesByMacAddress, err := net.detectMacAddresses()
+	if err != nil {
```

Comment:
Can you add a test for an error raised by `detectMacAddresses()`? It'd be great if you can test the same for `SetupNetworking()` which indirectly also uses `detectMacAddresses`.